src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -307,55 +435,53 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
-                {
-                    session.Properties[HasSuggestionItemOptions] = completionList.SuggestionModeItem != null;
-                }
+                session.Properties[HasSuggestionItemOptions] = completionList.SuggestionModeItem != null;
+            }
 
-                var excludedCommitCharacters = GetExcludedCommitCharacters(completionList.Items);
-                if (excludedCommitCharacters.Length > 0)
+            var excludedCommitCharacters = GetExcludedCommitCharacters(completionList.Items);
+            if (excludedCommitCharacters.Length > 0)
+            {
+                if (session.Properties.TryGetProperty(ExcludedCommitCharacters, out ImmutableArray<char> excludedCommitCharactersBefore))
                 {
-                    if (session.Properties.TryGetProperty(ExcludedCommitCharacters, out ImmutableArray<char> excludedCommitCharactersBefore))
-                    {
-                        excludedCommitCharacters = excludedCommitCharacters.Union(excludedCommitCharactersBefore).ToImmutableArray();
-                    }
-
-                    session.Properties[ExcludedCommitCharacters] = excludedCommitCharacters;
+                    excludedCommitCharacters = excludedCommitCharacters.Union(excludedCommitCharactersBefore).ToImmutableArray();
                 }
 
-                // We need to remember the trigger location for when a completion service claims expanded items are available
-                // since the initial trigger we are able to get from IAsyncCompletionSession might not be the same (e.g. in projection scenarios)
-                // so when they are requested via expander later, we can retrieve it.
-                // Technically we should save the trigger location for each individual service that made such claim, but in reality only Roslyn's
-                // completion service uses expander, so we can get away with not making such distinction.
-                if (!session.Properties.ContainsProperty(ExpandedItemTriggerLocation))
-                {
-                    session.Properties[ExpandedItemTriggerLocation] = triggerLocation;
-                }
+                session.Properties[ExcludedCommitCharacters] = excludedCommitCharacters;
+            }
+
+            // We need to remember the trigger location for when a completion service claims expanded items are available
+            // since the initial trigger we are able to get from IAsyncCompletionSession might not be the same (e.g. in projection scenarios)
+            // so when they are requested via expander later, we can retrieve it.
+            // Technically we should save the trigger location for each individual service that made such claim, but in reality only Roslyn's
+            // completion service uses expander, so we can get away with not making such distinction.
```

Comment:
i dont' undrestand the comment at all.  why/when would trigger location ever change?  isn't it always the same even if the user turns the expander off/on?

I would expect the expander only affects what we display, everything else is the same.

Reply:
My memory about this is a little foggy, so I have to debug it to be sure. But I think the comment above at least tells us why it's needed. In certain scenario (like cshtml), the intial trigger location we get from the IAsyncCompeltionSession (by casting it to [IAsyncCompletionSessionOperations](http://index/?leftProject=Microsoft.VisualStudio.Language&leftSymbol=p3a620tf1ygi&file=AsyncCompletion%5cIAsyncCompletionSessionOperations.cs)) is not the same as passed to us via `GetCompletionContextAsync`, and you'd notice that `GetExpandedCompletionContextAsync` doesn't pass in a trigger location at all.

Again, I don't recall why editor is implemented this way, I have to dig deeper to understand, which is what I plan to do as part of https://github.com/dotnet/roslyn/issues/59262